modules/trustxBidAdapter.js

```diff
@@ -0,0 +1,167 @@
+  var reqHost = '//sofia.trustx.org';
+  var reqPath = '/hb?';
+  var LOG_ERROR_MESS = {
+    noAuid: 'Bid from response has no auid parameter - ',
+    noAdm: 'Bid from response has no adm parameter - ',
+    noBid: 'Array of bid objects is empty',
+    noPlacementCode: 'Can\'t find placementCode for bid with auid - ',
+    havePCodeFor: ', placementCode is available only for the following uids - ',
+    emptyUids: 'Uids should be not empty',
+    emptySeatbid: 'Seatbid array from response has empty item',
+    emptyResponse: 'Response is empty',
+    hasEmptySeatbidArray: 'Response has empty seatbid array',
+    hasNoArrayOfBids: 'Seatbid from response has no array of bid objects - '
+  };
+
+  function _makeHandler(auids, placementMap) {
+    var cbName = bidderCode + '_callback_wrapper_' + auids.join('_');
+    $$PREBID_GLOBAL$$[cbName] = function(resp) {
+      delete $$PREBID_GLOBAL$$[cbName];
+      _responseProcessing(resp, auids, placementMap);
+    };
+    return '$$PREBID_GLOBAL$$.' + cbName;
+  }
+
+  function _sendRequest(auids, placementMap) {
+    var query = [];
+    var path = reqPath;
+    var needToSendStat = Math.random() < 0.1;
+    $$PREBID_GLOBAL$$.needToSendTrustxStat = needToSendStat;
+    query.push('u=' + encodeURIComponent(location.href));
+    query.push('auids=' + encodeURIComponent(auids.join(',')));
+    query.push('cb=' + _makeHandler(auids, placementMap));
+    query.push('pt=' + (window.globalPrebidTrustxPriceType === 'net' ? 'net' : 'gross'));
+    query.push('hbstat=' + (needToSendStat ? '1' : '0'));
+
+    adloader.loadScript(reqHost + path + query.join('&'));
+  }
+
+  function _callBids(params) {
+    var auids = [];
+    var placementMap = {};
+    var hasBid;
+    var bid;
+    var bids = params.bids || [];
+    for (var i = 0; i < bids.length; i++) {
+      bid = bids[i];
+      if (bid && bid.bidder === bidderCode && bid.placementCode) {
+        hasBid = true;
+        if (bid.params && bid.params.uid) {
+          if (!placementMap[bid.params.uid]) {
+            placementMap[bid.params.uid] = [bid.placementCode];
+            auids.push(bid.params.uid);
+          } else {
+            placementMap[bid.params.uid].push(bid.placementCode);
+          }
+        }
+      }
+    }
+
+    if (auids.length) {
+      _sendRequest(auids, placementMap);
+    } else if (hasBid) {
+      utils.logError(LOG_ERROR_MESS.emptyUids);
+    }
+  }
+
+  function _getBidFromResponse(resp) {
+    if (!resp) {
+      utils.logError(LOG_ERROR_MESS.emptySeatbid);
+    } else if (!resp.bid) {
+      utils.logError(LOG_ERROR_MESS.hasNoArrayOfBids + JSON.stringify(resp));
+    } else if (!resp.bid[0]) {
+      utils.logError(LOG_ERROR_MESS.noBid);
+    }
+    return resp && resp.bid && resp.bid[0];
+  }
+
+  function _forEachPlacement(error, bid, placementCode) {
+    var bidObject;
+    if (error) {
+      bidObject = bidfactory.createBid(2);
+    } else {
+      bidObject = bidfactory.createBid(1);
```

Comment:
pass through the bid request object as a 2nd argument to `createBid`. See example here: https://github.com/prebid/Prebid.js/blob/master/modules/appnexusAstBidAdapter.js#L206

Reply:
thanks for the update. do we need to apply these changes to the code by ourselves?

Reply:
Yes please make the update so we can merge your adapter. Thanks.